ming_sim/cli_backend.py

```
@@ -2197,6 +2197,28 @@ def extract_draft_intent(
     if correction_block and not correction_block.endswith("\n"):
         correction_block += "\n"
     stalled_push_facts = _stalled_deliberation_push_facts(db)
+    from ming_sim.action_clusters import (
+        assert_action_candidate_shape,
+        cluster_fields_prompt,
+        project_cluster_fields,
+    )
+    grant_fields_prompt = cluster_fields_prompt("grant_allocation")
+
+    def _normalize_grant_transport(value: Mapping[str, Any]) -> Dict[str, Any]:
```

Ruff (0.16.2)

[error] Undefined name `Mapping`

(F821)